internal/rbd/snapshot.go

```diff
@@ -89,7 +89,16 @@ func generateVolFromSnap(rbdSnap *rbdSnapshot) *rbdVolume {
 	vol.RadosNamespace = rbdSnap.RadosNamespace
 	vol.RbdImageName = rbdSnap.RbdSnapName
 	vol.ImageID = rbdSnap.ImageID
-	return vol
+
+	if rbdSnap.isEncrypted() {
+		vol.conn = rbdSnap.conn.Copy()
+		err := rbdSnap.copyEncryptionConfig(&vol.rbdImage)
```

Comment:
This is wrong for the case where the KMS uses the VolID for encrypting.decrypting the passphrase (SecretMetadataKMS). The encryption of the passphrase for the new rbdVolume is done with an incorrect VolID.